### DISCUSSION TITLE
statically generated site - adding 404 page that doesn't require API calls

### DISCUSSION THREAD

**Discussion by lukemorell** [nuxt/nuxt#23243] [Questions]
So we have a Nuxt project (using version ^3.6.5).

`ssr: true` in the `nuxt.config.ts` file.

We have a basic `error.vue` file in the project root to handle errors.

We also have a `[[slug]].vue` file inside the `/pages` folder which makes an API call to get content from the CMS, using the params slug.
(e.g. `example.com/about`, `example.com/contact`)

_[[slug]].vue_:
```
<script setup>
import pageQuery from '~/apollo/queries/page.js'
const route = useRoute()
const slug = route.params.slug ? route.params.slug : 'home'
const { data } = await useAsyncQuery(pageQuery, { slug })
</script>

<template>
  <StructurePage
    v-if="data?.page?.data"
    :page="data.page.data.attributes"
  />
</template>
```

The static site that is generated - we are serving as flat html files using apache. If a html page doesn't exist, we want to serve a basic 404 page with apache:
```ErrorDocument 404 /404```

However, Nuxt is trying to be dynamic (making API calls) when visiting a page that doesn't exist (e.g. `/abcxyz`).
How do we get Nuxt to behave fully static? (i.e. on the production site, no API calls made to the CMS)

Have checked lots of forums, but not managed to find the solution we need yet. (managed to have this setup working perfectly with Nuxt2, but Nuxt3 is trying to be hybrid)

**Comment by TheAlexLichter** [comment]
AFAIK you can't prerender error pages for now.
Follow https://github.com/nuxt/nuxt/issues/18718 ☺️

**Reply by lukemorell** [reply]
Thanks @manniL - some helpful answers in there about the `error.vue` and also layouts within that file. Will follow this further

**Comment by codeflorist** [comment]
this seems to be a duplicate of #22952.

check out my comment for a kind-of-solution: https://github.com/nuxt/nuxt/issues/22952#issuecomment-1722162867

though you probably won't like it.
